src/Scientist/Internals/GCHelper.cs

```diff
@@ -0,0 +1,132 @@
+﻿namespace GitHub.Internals
+{
+    using System;
+    using System.Reflection;
+    using System.Runtime.InteropServices;
+
+    internal static class GCHelper
+    {
+        private const string DebugConfigurationName = "DEBUG";
+        internal const string ReleaseConfigurationName = "RELEASE";
+        internal const string Unknown = "?";
+
+        public static bool IsMono { get; } = Type.GetType("Mono.Runtime") != null;
+
+        public static bool IsFullFramework =>
+#if (NET451)
+            true;
+#else
+            System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);
+#endif
+
+        public static bool IsNetCore =>
+#if (NET451)
+            false;
+#else
+            System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase);
+#endif
+
+        private static long AllocationQuantum { get; } = CalculateAllocationQuantumSize();
+
+        private static readonly Func<long> GetAllocatedBytesForCurrentThreadDelegate = GetAllocatedBytesForCurrentThread();
+
+        /// <summary>
+        /// returns total allocated bytes (not per operation)
+        /// </summary>
+        /// <param name="excludeAllocationQuantumSideEffects">Allocation quantum can affecting some of our nano-benchmarks in non-deterministic way.
+        /// when this parameter is set to true and the number of all allocated bytes is less or equal AQ, we ignore AQ and put 0 to the results</param>
+        /// <returns></returns>
+        public static long GetAllocatedBytes()
+        {
+            var allocation = GetAllocatedBytesByRuntime();
+
+            if (IsNetCore)
+            {
+                return allocation;
+            }
+
+            return allocation <= AllocationQuantum ? 0L : allocation;
+        }
+
+        private static long GetAllocatedBytesByRuntime()
+        {
+            // Monitoring is not available in Mono, see http://stackoverflow.com/questions/40234948/how-to-get-the-number-of-allocated-bytes-
+            if (IsMono)
+            {
+                return 0;
+            }
+            //GC.Collect();
+
+#if (NET451)
+            // "This instance Int64 property returns the number of bytes that have been allocated by a specific 
+            // AppDomain. The number is accurate as of the last garbage collection." - CLR via C#
+            // so we enforce GC.Collect here just to make sure we get accurate results
+            return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+#elif (NETSTANDARD2_0)
+            // it can be a .NET app consuming our .NET Standard package
+            if (IsFullFramework) 
```

Comment:
What do you think about taking this check outside of this method, and use it during the creation of `GetAllocatedBytesForCurrentThreadDelegate`? We'd be able to save the check, and then always just run
```csharp
GC.Collect();
return AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
```

Reply:
So do it purely on #IfDef? The reason we do a runtime framework check is because the if you used the code within a dotnet Standard library you wouldn't know at compile time your runtime (FX, Core or Mono). Apologies if I've missed your point here I'm not convinced that GitHub is pointing me to the right line.

Reply:
Yeah - this may have been a little misleading. What i'm suggesting is moving this logic out to where we construct the `GetAllocatedBytesForCurrentThreadDelegate` delegate. I'll add some comments in that area to help facilitate.

Reply:
See below, I think this is addressed with your later comment.